Assignments/Program.cs

```diff
@@ -1,3 +1,202 @@
+// Academy23 Assignments
+// Copyright (c) Metamation India.
+// ---------------------------------------------------------------------------------------
+// Program.cs
+// A12 - Wordle Game
+// ---------------------------------------------------------------------------------------
 
-Console.WriteLine("Hello, World!");
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.IO;
+using static System.ConsoleColor;
+using System.Reflection;
+
+namespace Academy;
+public class A12 {
+   public static void Main () => new Wordle ().PlayTheGame ();
+}
+
+public class Wordle {
+   public string secretWord, text;
+   public int triesCount = 0, ithComment = 0;
+   public bool testing = false, gameOver;
+   public int originX, originY;               // Reference X & Y coordinates.
+   public List<List<char>> word = new (6);    // List To display and store user entered letters.
+   readonly List<char> grayList = new (),     // List to collect incorrect letters.
+                       blueList = new (),     // List to collect correct but misplaced letters.
+                       greenList = new ();    // List to collect correct and correctly placed letters.
+   public string[] dictWords, guessWords;     // To store loaded strings
+   public StreamWriter testOutputFile = null; // Created for testing purpose.
+
+   public void PlayTheGame () {
+      SetTheBoard ();
+      DisplayTheBoard ();
+      while (!gameOver) {
+         ConsoleKeyInfo key = Console.ReadKey (true);
+         UpdateGameState (key.Key);
+         DisplayTheBoard ();
+      }
+      Console.CursorTop = 28;
+   }
+
+   /// <summary>Initializes the game</summary>
+   public void SetTheBoard () {
+      text = "";
+      (originX, originY) = testing ? (49, 3) : (Console.CursorLeft + (Console.WindowWidth / 2) - 13, Console.CursorTop);
+      dictWords = LoadStrings ("dict-5.txt");
+      guessWords = LoadStrings ("puzzle-5.txt");
+      Console.OutputEncoding = Encoding.Unicode;
+      if (!testing) Console.CursorVisible = false;
+      // Creating Type Board elements:
+      for (int i = 0; i < 6; i++)
+         word.Add (new List<char> { '\u00b7', '\u00b7', '\u00b7', '\u00b7', '\u00b7' });
+      // Generating a random 5 letter word from loaded strings file.
+      var r = new Random ();
+      secretWord = guessWords[r.Next (guessWords.Length)];
+   }
+
+   /// <summary>Displays the updated game board</summary>
+   public void DisplayTheBoard () {
+      if (testing) {
+         // TYPING BOARD:
+         testOutputFile.Write (string.Join ("", Enumerable.Repeat ("\r\n", 3)));
+         if (text.Length == 1) testOutputFile.WriteLine ($"Try-{triesCount + 1}");
+         testOutputFile.Write (new string (' ', originX));
+         for (int i = 0; i < word.Count; i++) {
+            for (int j = 0; j < word[i].Count; j++) {
+               var (a, b, c) = (' ', ' ', word[i][j]);

```

Comment:
These for loops are common in both conditions; for testing and while running the application. Can you simplify this?